### PR TITLE
non-production: update README with histogram#time

### PR DESCRIPTION
**Pull request by ColinDKelley**
### Changed
- Update README to document metrics types, values, and supported methods.

### Deprecated
- Deprecated `gauge.set` and `histogram.add` default values of 1 in favor of explicit values.
These defaults will be dropped in version 2.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,65 @@
...
+
+After declaring the metrics, you may use `prometheus_metrics.<metric_name>` (in either the class or instance) to publish the metric.
+```ruby
+prometheus_metrics.request_total.increment       # increment the counter by 1
+
+prometheus_metrics.request_total.increment(5)    # increment the counter by 5
+
+prometheus_metrics.available_memory.set(10.25)   # set the gauge to 10.25
+
+prometheus_metrics.processing_time.add(100)      # Increment the bucket closest to the value 100
```
Comment (ColinDKelley) [inline README.md:59]: This was incorrectly documented as `observe` before.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,65 @@
...
+prometheus_metrics.processing_time.time { work } # measure the elapsed milliseconds to run the given block and increment the bucket closest to that time period in milliseconds
```
Comment (ColinDKelley) [inline README.md:60]: This wasn't documented at all before.

**Review by yinonrousso — COMMENTED** [review]
Thanks for getting this change in! Cool implementation of deprecating the method param default 👍 The documentation is clear and comprehensive 💯
✅ With merge conflicts resolved

**Comment by ColinDKelley** [comment]
@yinonrousso 

> Cool implementation of deprecating the method param default

Thanks! I thought it was kind of fun to use a symbol `:'1'` to represent the default of `1`. The alternate approach I've done before was to be more explicit with a default of `:__deprecated_default__` or something like that. That worked equally well. (I definitely didn't want `nil` since that could wind up getting sent in unintentionally.)

**Pull request merged**